### PR TITLE
feat(default-theme): Improve SwImage component with srcset

### PR DESCRIPTION
**Pull request by quando1910**
this PR closes https://github.com/vuestorefront/shopware-pwa/issues/1884
## Changes
- add new prop `srcset` for SwImage component.
- update srcset data of SwImage inside CmsElementImage component.

### Checklist
- [x] implement computed property for src set
- [x] find all occurances of SwImage component used in default-theme
- [x] pass src set extracted from available data from the parent (ignore if it's not possible).
- [x] use breakpoints set as default coming with media object (400, 800, 1200?).

### CODE REVIEW

File: packages/default-theme/src/cms/elements/CmsElementImage.vue [hunk 1]
```diff
@@ -1,13 +1,13 @@
 <template>
-  <div v-if="imgUrl">
+  <div v-if="imagesUrl && imagesUrl.length">
```
Comment (patzick) [inline packages/default-theme/src/cms/elements/CmsElementImage.vue:2]: it's a good practice to use optional chaining :)
```suggestion
  <div v-if="imagesUrl?.length">
```

File: packages/default-theme/src/cms/elements/CmsElementImage.vue [hunk 2]
```diff
@@ -57,8 +57,8 @@ export default {
       return this.content && this.content.data && this.content.data.media
     },
 
-    imgUrl() {
-      return getResizedImage(this.getMedia && this.getMedia.url)
+    imagesUrl() {
+      return this.getMedia ? this.getMedia.thumbnails : []
```
Comment (patzick) [inline packages/default-theme/src/cms/elements/CmsElementImage.vue:61]:
```suggestion
      return this.getMedia?.thumbnails ?? []
```

File: packages/default-theme/src/components/atoms/SwImage.vue [hunk 3]
```diff
@@ -11,13 +11,39 @@ export default {
...
   setup(props, { root }) {
     const width = (props.imageWidth && props.imageWidth + "px") || "100%"
     const height = (props.imageHeight && props.imageHeight + "px") || "100%"
     return {
       placeholder: getPlaceholderImage(width, height),
     }
   },
+  computed: {
+    srcsets () {
+      return this.srcset ? this.srcset.map(item => ({
```
Comment (patzick) [inline packages/default-theme/src/components/atoms/SwImage.vue:30]: let's move these computed properties to setup method, example:
```ts

const srcsets = computed(() => props.srcset?.map(item => ({
   src: item.url,
   width: item.width,
   resolution: item.width
  }) ?? []
)

```

**Review by patzick — CHANGES_REQUESTED** [review]
good start! :)
take a closer look at Optional Chaining and look more closely at Vue's Composition API, so we'll have logic inside the `setup` method

File: packages/default-theme/src/cms/elements/CmsElementImage.vue (hunk 1 above)
Comment (quando1910) [reply packages/default-theme/src/cms/elements/CmsElementImage.vue:2]: `<div v-if="imagesUrl?.length">` this code can not work, the terminal always said `Compiled with some errors`. I think the app does not allow us to use optional chaining in `v-if`.

Comment (patzick) [reply packages/default-theme/src/cms/elements/CmsElementImage.vue:2]: alright! So let's leave it like that! :)

**Pull request merged**